components/DiscussionsCard/DiscussionsCard.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import styles from './discussionsCard.module.scss'
+import {
+  ReplyIcon,
+  ThumbsupIcon,
+  ThumbsdownIcon,
+  TrashIcon,
+  PencilIcon
+} from '@primer/octicons-react'
+
+export type DiscussionsCardProps = {
+  discussionType: 'main' | string
```

Comment:
@HS-90 Is there any potential that there will be more than 2 possible values for `discussionType`? Also, what is the reason for different styling options? Do we have different styling on different pages for DiscussionCard component?

Reply:
I'm thinking that if we know there will never be more than 2 possible values for `discussionType`, then we might as well make the prop a boolean value, ex. `subMode?: boolean`

Reply:
If theres high potential that discussionType will have more than 2 possible values, id suggest we use a mapping approach instead of using ternary operators(see lines 39 and 42 for example)

Reply:
Example of mapping approach: 

```
className={styles[`card_body_${discussionType}]}
```

Reply:
The above might be a bad example.. im not sure its a good idea for name of the`class` to be part of the logic in this kind of way.

A more simple approach would probably be to have `main` and `sub` as keys in a hashmap, and then have as its values the corresponding styles for them

Reply:
> If theres high potential that discussionType will have more than 2 possible values, id suggest we use a mapping approach instead of using ternary operators(see lines 39 and 42 for example)

There should be only 2 types, one is a main card, which will be the original comment or thread, then replies will be the sub design, branching off of the original comment. 

A subcard can also have subcards branching off of them, kind of like how reddit works. but the design should be the same for every subcard, if that makes sense. I think I agree with your boolean approach

Reply:
If the only purpose of having a boolean is to check if a reply is the main one, wouldn't be better to instead check if the subreply doesn't have any parent comment and style accordingly?
Also, should we consider that any subreply that has n replies would be considered a main? if so, wouldn't that make things a bit confusing?
We didn't get to this point in the design meetings.

Reply:
> If the only purpose of having a boolean is to check if a reply is the main one, wouldn't be better to instead check if the subreply doesn't have any parent comment and style accordingly?

Why is it better?

Reply:
After a bit of though, it seems like it won't make any difference in our case, the only case it would be useful is if we want to style subthread replies with different levels/shades of colors.